### DISCUSSION TITLE
Connecting to existing block and moving existing block with new dagre designer problem

### DISCUSSION THREAD

**Discussion by tomy2105** [elsa-workflows/elsa-core#918] [Q&A]
Latest elsa from code and dagree designer....

In previous desinger you could click and drag from + on some block outcome to an existing block and it would connect this outcome to that existing block.

In new dagree designer this is not possible because click and drag does scrolling.
How do you connect to an existing block in new designer? 

Same question goes for moving existing block to new position.
Previously you could drag existing block and drop it on any (preferably unconnected outcome +) and it would be placed there.
Again with new designer not possible due to drag and drop used for scrolling.
Is it possible any other way?

**Comment by sfmskywalker** [comment]
It is possible, but it can use some further improvements.

To connect an outcome to an already existing activity, you need to press and hold the SHIFT button when hovering your mouse cursor on an outcome (on the PLUS icon) and CLICK. Then release the SHIFT button and CLICK on the target activity to connect to.

Here's what it would look like:

![click-connect](https://user-images.githubusercontent.com/938393/116568715-fe869f00-a908-11eb-8f1a-95c84875fabd.gif)

**Reply by tomy2105** [reply]
OK, not ideal but connecting can be done.
How about moving existing block to new position in workflow?

**Reply by sfmskywalker** [reply]
That's something we should still try to implement, at some point. The first thing we'd have to change is the "automatic" panning as you drag the screen I think. Perhaps panning should be activated only when pressing space bar. Then it might be easier to capture drag/drop events from dagre-d3 when trying to drag a node.
